src/service/point-of-sale-service.ts

```diff
@@ -163,50 +128,23 @@ export default class PointOfSaleService {
   ): Promise<PaginatedPointOfSaleResponse> {
     const { take, skip } = pagination;
 
-    const results = await Promise.all([
-      (await this.buildGetPointsOfSaleQuery(filters, user)).limit(take).offset(skip).getRawMany(),
-      (await this.buildGetPointsOfSaleQuery(filters, user)).getCount(),
-    ]);
-
-    let records;
-    if (filters.returnContainers) {
-      const pointOfSales: PointOfSaleWithContainersResponse[] = [];
-      await Promise.all(results[0].map(
-        async (rawPointOfSale) => {
-          pointOfSales.push(
-            await this.asPointOfSaleResponseWithContainers(
-              this.asPointOfSaleResponse(rawPointOfSale),
-            ),
-          );
-        },
-      ));
-      records = pointOfSales;
-    } else {
-      records = results[0].map((rawPointOfSale) => this.asPointOfSaleResponse(rawPointOfSale));
-    }
+    const [data, count] = await PointOfSaleRevision.findAndCount({ ...(await this.getOptions(filters, user)), take, skip });
 
+    const records = data.map((revision: PointOfSaleRevision) => this.revisionToResponse(revision));
     return {
       _pagination: {
-        take, skip, count: results[1],
+        take, skip, count,
       },
       records,
     };
   }
 
   /**
-   * Turns an PointOfSaleRevision into an PointOfSaleResponse
+   * Updates a PointOfSale
+   * @param update - The update to apply
    */
-  static toPointOfSaleResponse(pointOfSale: PointOfSaleRevision)
-    : PointOfSaleResponse {
-    return {
-      name: pointOfSale.name,
-      owner: parseUserToBaseResponse(pointOfSale.pointOfSale.owner, false),
-      id: pointOfSale.pointOfSale.id,
-      revision: pointOfSale.revision,
-    } as PointOfSaleResponse;
-  }
-
-  public static async applyPointOfSaleUpdate(base: PointOfSale, update: BasePointOfSaleParams) {
+  public static async updatePointOfSale(update: UpdatePointOfSaleParams) {
```

Comment:
This function still has an `|` in its return type (`PointOfSaleResponse` or `PointOfSaleWithContainersResponse`). Please make explicit which type is returned, as this does not seem to be a choice for the user.